language-server/test/dotty/tools/languageserver/util/actions/CodeCompletion.scala

```diff
@@ -0,0 +1,28 @@
+package dotty.tools.languageserver.util.actions
+
+import dotty.tools.languageserver.util.PositionContext
+import dotty.tools.languageserver.util.embedded.CodeMarker
+import dotty.tools.languageserver.util.server.TestFile
+
+import scala.collection.JavaConverters._
+
+class CodeCompletion(val marker: CodeMarker, completions: List[(String, String, String)]) extends ActionOnMarker {
+
+  override def execute(): Exec[Unit] = {
+    val res = server.completion(marker.toTextDocumentPositionParams).get()
+    assert(res.isRight, res)
+    val cList = res.getRight
+    assert(!cList.isIncomplete, res)
+    completions.foreach { completion =>
```

Comment:
If you convert everything into set you should be able to check that one set is contained in another.